emulators/mame/Portfile

```diff
@@ -14,52 +14,69 @@ PortSystem          1.0
-    electronic technology continues to rush forward, MAME prevents this important \
-    "vintage" software from being lost and forgotten. This is achieved by documenting \
-    the hardware and how it functions. The source code to MAME serves as this \
+long_description    \
+    The purpose of MAME is to preserve decades of software history. As electronic \
+    technology continues to rush forward, MAME prevents this important "vintage" \
+    software from being lost and forgotten. This is achieved by documenting the \
+    hardware and how it functions. The source code to MAME serves as this \
     documentation.
-
-github.tarball_from archive
+homepage            https://www.mamedev.org
 
 #------------------------------------------------------------------------------
-# Patches and Checksums
+# Versions, Patches, and Checksums
 #------------------------------------------------------------------------------
 
-# Patch 'language-portuguese_brazil' applies to release 0.226; fixes
-# errors during translation build phase.
-# Fix submitted, and merged, by Mame maintainers; will be part of Mame
-# releases from 0.227 onward. Tracked by Mame issue:
-# https://github.com/mamedev/mame/issues/7510
-#
-# Patch 'src-posixfile-build-error' applies to release 0.226; fixes
-# compilation errors for MacOS 10.8 and Xcode 5.x.
-# Fix submitted, and merged, by Mame maintainers; will be part of Mame
-# releases from 0.227 onward. Tracked by Mame issue:
-# https://github.com/mamedev/mame/issues/7536
-#
-# Patch 'scripts-genie-compile-warnings' applies to all releases; greatly
-# reduces compilation warnings.
-
-patchfiles          mame-patch-0226-language-portuguese_brazil.diff \
-                    mame-patch-0226-src-posixfile-build-error.diff \
-                    mame-patch-0226-scripts-genie-compile-warnings.diff
-
-checksums           rmd160  b9e4ae321b7673790d374c63bbe966d1502d6738 \
+if {${name} eq ${subport}} {
+    revision        3
+    conflicts       ${name}-latest
+
+    set g_mame_release \
+                    "0226"
+
+    # Patch 'language-portuguese_brazil' applies to release 0.226; fixes
+    # errors during translation build phase.
+    # Merged into Mame releases from 0.227 onward. Tracked by issue:
+    # https://github.com/mamedev/mame/issues/7510
+    #
+    # Patch 'src-posixfile-build-error' applies to release 0.226; fixes
+    # compilation errors for MacOS 10.8 and Xcode 5.x.
+    # Merged into Mame releases from 0.227 onward. Tracked by issue:
+    # https://github.com/mamedev/mame/issues/7536
+    patchfiles      \
+                    mame-patch-0226-language-portuguese_brazil.diff \
+                    mame-patch-0226-src-posixfile-build-error.diff
+
+    checksums       \
+                    rmd160  b9e4ae321b7673790d374c63bbe966d1502d6738 \
                     sha256  7c4c9ec232ba988e65fd29665c9b8e40b5ac3aa9f561eeb107cebbf08ba94baf \
                     size    196379874
+}
+
+subport ${name}-latest {
```

Comment:
In Macports, we call such a port that tracks a newer version than the latest `-devel`. So let's call this one `mame-devel`.

Reply:
Well, the `-latest` suffix was used, as that's what it truly is: The latest version. It's not a development version, but rather, the latest released version.

This is necessary, as Mame 0.227 now depends on C++ 2017 features, the most notable being aligned allocations. And the latter isn't supported by the system libc++ prior to macOS 10.13. (Though when we support a replacement libc++ in MacPorts - hopefully sooner, rather than later - that won't be a problem any longer.)

@kencu Ken and I have discussed this at length in ticket [61976 - mame: update to 0.227](https://trac.macports.org/ticket/61976).

I'm certainly open to a different naming convention, but should a release version use a `-devel` suffix...?

Reply:
In that case, a release version would not have a suffix and should simply be called `mame`.

Reply:
> In that case, the latest release version would not have a suffix and should simply be called `mame`. Older release versions are named with the version number.

That was one of the ideas I originally considered. The problem is, users who already have Mame installed - v0.226 - will then be prompted to upgrade the port. But since the new release (0.227) requires macOS 10.13+, while 0.226 builds and runs on macOS releases from 10.6 through 10.15, the port will be flagged as outdated - and an update attempted every time those users run a `port upgrade outdated` - but the port can only be upgraded for users running on macOS 10.13+. And the upgrade will fail with an error message, every time they run it.

That's going to become very annoying, very quickly, for all of our users on older macOS releases. And asking them to workaround it by doing a `port upgrade outdated and not mame` each time, seems less-than-ideal.

I suppose we could add logic to the portfile, and prompt the affected users to switch to `mame-0226`. But I'd really prefer not to go down that path.

In short, the `mame-latest` subport route was chosen, to provide a user-friendly upgrade path... and ensure users of 0.226 [on legacy macOS releases] don't have to deal with any pain or annoyance. All whilst still providing a pathway for users of macOS 10.13+, to use newer Mame releases.

Does that make sense?

Reply:
Oh, I'm aware of the portfile techniques that we could use, in terms of port replacement/renaming, etc. But again, I'm concerned about the user experience...

Reply:
Hmmm. Pondering this a bit more, we could simply eliminate the subport, and automatically set the version based on the macOS release the user is running on. So folks on macOS 10.13+ would see 0.227, while folks on older macOS releases would see 0.226.

This would certainly provide a seamless user experience. Though it's also nice to show that we support newer releases somewhere, even if it's only in a subport.

Thoughts?

Reply:
@kencu Adding Ken to the mix, since we discussed this a bit in the ticket...

Reply:
So I'm afraid I'm with Frank here.

We should just have one port named "mame". 

On current systems, the version can be the newest available. On older systems, the version can be the last one that could build.

But it is overly confusing to have an extra port named "mame-latest" that people need to know to install when they want to install mame.

You are right that there is an ld64-latest port. To be fair there, though, that is not a user-installed port in 99.99% of cases. It is automatically installed by the "ld64" metaport, and then symlinked in. So that is a bit of a special case.

What we're trying to do is make this as simple as possible for users. Somebody two years from now is going to want to install "mame" and will get either an old version, or confused. So it's best to go with the MacPorts' standard ways of doing things for consistency.

Reply:
Frank, I edited my prior comments down to the bare minimum, to ease the discussion.

Also, per Ken's suggestion, I added text to the description and notes for both ports. This provides the user with an explanation regarding what macOS release is supported for each, as well as mentioning the opposite port. In short, it helps to eliminate any confusion for the user.

Let me know if you're open to the proposed naming, particularly with that additional text added.

Reply:
Given that a single port provides a seamless user experience, I'm good with that. Thanks guys!

I'll work on the changes, and let you folks know when they're ready.

Reply:
Changes complete.